### ISSUE TITLE
Influx apply creates duplicate Tasks and Notification Rules.

### ISSUE THREAD

**Issue by ryanrolds** [influxdata/influxdb#24092]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Create a Task
2. Export template 
    * `influx export all -f template.yaml`
3. Apply template (previews show the task as being added, not unchanged/replaced)
    * `influx apply -f template.yaml`

Seeing the same behavior with Notification Rules.

> I'm happy to work up a PR if we agree this is a bug.

__Expected behavior:__

A single copy of the task exists, and the preview shows unchanged or replaced.

__Actual behavior:__

A duplicate task is created.

__Environment info:__

* System info: Run `uname -srm` and copy the output here
    * Linux 5.10.102.1-microsoft-standard-WSL2 x86_64
* InfluxDB version: Run `influxd version` and copy the output here
    * InfluxDB v2.6.1 (git: 9dcf880fe0) build_date: 2022-12-29T15:53:07Z
    * Influx CLI 2.6.1 (git: 61c5b4d) build_date: 2022-12-29T15:41:09Z
* Other relevant environment details: Container runtime, disk info, etc
    * Docker image: influxdb:2.6-alpine

__Config:__

Running using Docker Compose and running the CLI commands by execing into the running container.

**Comment by jeffreyssmith2nd** [comment]
Hey @ryanrolds, I do agree that I would expect to see "unchanged" for applying back the same resources I already have. This seems to work as expected for buckets, though not for others. PRs are always welcome! The API endpoint that is being called for `influx apply` is [here](https://github.com/influxdata/influxdb/blob/f74c69c5e4a9c865c204949f3bb85cf58c56c7b6/pkger/http_server_template.go#L329), that would be a good place to start investigating.

**Comment by ryanrolds** [comment]
Thanks. I will take a look and try to work something over the next week or so.

**Comment by ryanrolds** [comment]
It looks like the model for Buckets fetches by name when an ID isn't present, and templates do not include the ID. I'm under the impression that names are not unique for all models (as I've seen Tasks, and others, have entities with the same name). Would exporting the id along with the name and only looking up existing entities by ID be acceptable? 

If we have to fetch by name, not id, then a policy on handling multiple entities with the same name while importing will need to be decided.

Concerns:
* Templates that include multiple entities of the same type with the same name.
  *  How do we know we are updating the right one without a unique id?
  * Should these entities be rejected?
* When importing a single entity and multiple entities of that type with the same are found.
  * Which one is updated?
  * How do we know that is the one the user expected to be updated?
  * Should the entity be rejected?

IMHO, not fetching+merging by name and only using a unique ID is the best path forward, but I don't have the business context/concerns that drive the current implementation.

**Comment by jeffreyssmith2nd** [comment]
I agree with your suggestion, the ID is the only thing that is guaranteed to be unique on a lot of these models. That's what should be used for export/import and determining existence or not.

Thanks again for taking a look at this!

**Comment by ryanrolds** [comment]
Is there any concerns with the shared templates? If people have already imported a template without an id (name instead) what is the expected behavior when importing a new version (is this even a thing?) of the template that has IDs?

**Comment by jeffreyssmith2nd** [comment]
I think when importing a template with ids, it should use that to validate uniqueness. If there are not ids in the template, then we should fall back to the current behavior to prevent potentially breaking things.

I think that will handle things okay? I'm not super familiar with templates or their usage patterns so I don't know how often people are applying the same template over and over again unless testing?

**Comment by LCerebo** [comment]
Hi, any update on the argument?
I'm looking forward to use influx apply with ansible where the idempotency is almost a necessity.

**Comment by ryanrolds** [comment]
I looked at doing the work, but the API would need significant rework and I abandoned the idea. I just put up with the wonky behavior and hope they fix it in the next version.

**Comment by nykopol** [comment]
Hello, I run across the same issue. 
I need a idempotente behavior for a complexe monitoring where the desired config is version controlled with git. 
Is there any plan to work on this issue or at leas a workaround ?

**Comment by ryanrolds** [comment]
I recall that the issue's crux is that the identifiers could be more consistent. What ends up in the files that would get imported don't have the right IDs. 

A workaround will likely involve a script that creates any missing data in the right order and with the right IDs.

**Comment by impresionista** [comment]
Hello people. Does anyone found some workaround for this?
